File: src/main/java/io/github/chaosdave34/kitpvp/kits/impl/kitpvp/PoseidonKit.java
package io.github.chaosdave34.kitpvp.kits.impl.kitpvp;

import io.github.chaosdave34.kitpvp.items.CustomItemHandler;
import io.github.chaosdave34.kitpvp.kits.Kit;
import org.bukkit.Color;
import org.bukkit.Material;
import org.bukkit.enchantments.Enchantment;
import org.bukkit.inventory.ItemStack;

public class PoseidonKit extends Kit {
    public PoseidonKit() {
        super("poseidon", "Poseidon");
    }

    @Override
    public ItemStack getHeadContent() {
        ItemStack turtleHelmet = new ItemStack(Material.TURTLE_HELMET);
        turtleHelmet.addEnchantment(Enchantment.PROTECTION_ENVIRONMENTAL, 2);
        turtleHelmet.addEnchantment(Enchantment.OXYGEN, 3);
        turtleHelmet.addEnchantment(Enchantment.WATER_WORKER, 1);
        return turtleHelmet;
    }

    @Override
    public ItemStack getChestContent() {
        ItemStack leatherChestplate = new ItemStack(Material.LEATHER_CHESTPLATE);
        leatherChestplate.addEnchantment(Enchantment.PROTECTION_ENVIRONMENTAL, 2);
        setLeatherArmorColor(leatherChestplate, Color.AQUA);
        return leatherChestplate;
    }

    @Override
    public ItemStack getLegsContent() {
        ItemStack leatherLeggings = new ItemStack(Material.LEATHER_LEGGINGS);
        leatherLeggings.addEnchantment(Enchantment.PROTECTION_ENVIRONMENTAL, 2);
        setLeatherArmorColor(leatherLeggings, Color.AQUA);
        return leatherLeggings;
    }

    @Override
    public ItemStack getFeetContent() {
        ItemStack leatherBoots = new ItemStack(Material.LEATHER_BOOTS);
        leatherBoots.addEnchantment(Enchantment.PROTECTION_ENVIRONMENTAL, 2);
        leatherBoots.addEnchantment(Enchantment.DEPTH_STRIDER, 3);
        setLeatherArmorColor(leatherBoots, Color.AQUA);
        return leatherBoots;
    }

    @Override
    public ItemStack[] getInventoryContent() {
        ItemStack riptideTrident = new ItemStack(Material.TRIDENT);
        riptideTrident.addUnsafeEnchantment(Enchantment.RIPTIDE, 5);

        return new ItemStack[]{
                CustomItemHandler.POSEIDONS_TRIDENT.build(),
                riptideTrident,
                new ItemStack(Material.WATER_BUCKET),
        };
    }

    @Override
    public ItemStack[] getKillRewards() {
        return new ItemStack[]{
                new ItemStack(Material.GOLDEN_APPLE),
                new ItemStack(Material.PRISMARINE, 32),
        };
    }
}

File: src/main/kotlin/io/github/chaosdave34/kitpvp/kits/impl/kits/PoseidonKit.kt
package io.github.chaosdave34.kitpvp.kits.impl.kits

import io.github.chaosdave34.kitpvp.items.CustomItemHandler
import io.github.chaosdave34.kitpvp.kits.Kit
import io.github.chaosdave34.kitpvp.ultimates.Ultimate
import io.github.chaosdave34.kitpvp.ultimates.UltimateHandler
import org.bukkit.Color
import org.bukkit.Material
import org.bukkit.enchantments.Enchantment
import org.bukkit.inventory.ItemStack

class PoseidonKit : Kit("poseidon", "Poseidon") {

    override fun getHeadContent(): ItemStack {
        val turtleHelmet = ItemStack.of(Material.TURTLE_HELMET)
        turtleHelmet.addEnchantment(Enchantment.PROTECTION, 2)
        turtleHelmet.addEnchantment(Enchantment.RESPIRATION, 3)
        turtleHelmet.addEnchantment(Enchantment.AQUA_AFFINITY, 1)
        return turtleHelmet
    }

    override fun getChestContent(): ItemStack {
        val chestplate = ItemStack.of(Material.LEATHER_CHESTPLATE)
        chestplate.addEnchantment(Enchantment.PROTECTION, 2)
        chestplate.setLeatherArmorColor(Color.AQUA)
        return chestplate
    }

    override fun getLegsContent(): ItemStack {
        val leggings = ItemStack.of(Material.LEATHER_LEGGINGS)
        leggings.addEnchantment(Enchantment.PROTECTION, 2)
        leggings.setLeatherArmorColor(Color.AQUA)
        return leggings
    }

    override fun getFeetContent(): ItemStack {
        val boots = ItemStack.of(Material.LEATHER_BOOTS)
        boots.addEnchantment(Enchantment.PROTECTION, 2)
        boots.addEnchantment(Enchantment.DEPTH_STRIDER, 3)
        boots.setLeatherArmorColor(Color.AQUA)
        return boots
    }

    override fun getInventoryContent(): Array<ItemStack?> {
        val trident = ItemStack.of(Material.TRIDENT)
        trident.addUnsafeEnchantment(Enchantment.LOYALTY, 10)
        trident.addUnsafeEnchantment(Enchantment.CHANNELING, 5)

        val riptideTrident = ItemStack.of(Material.TRIDENT)
        riptideTrident.addEnchantment(Enchantment.RIPTIDE, 3)

        return arrayOf(
            trident,
            riptideTrident,
            CustomItemHandler.WATER_BURST.build(),
            ItemStack.of(Material.WATER_BUCKET),
        )
    }

    override fun getKillRewards(): Array<ItemStack> {
        return arrayOf(
            ItemStack.of(Material.GOLDEN_APPLE),
            ItemStack.of(Material.PRISMARINE, 32),
        )
    }

    override fun getUltimate(): Ultimate = UltimateHandler.STORM
}